examples/preprocessing/plot_discretization.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+"""
+================================================================
+Using KBinsDiscretizer to discretize continuous features
+================================================================
+
+The example compares prediction result of linear regression (linear model)
+and decision tree (tree based model) with and without discretization of
+real-valued features.
+
+As is shown in the result before discretization, linear model is fast to
+build and relatively straightforward to interpret, but can only model
+linear relationships, while decision tree can build a much more complex model
+of the data. One way to make linear model more powerful on continuous data
+is to use discretization (also known as binning). In the example, we
+discretize the feature and one-hot encode the transformed data. Note that if
+the bins are not reasonably wide, there would appear to be a substantially
+increased risk of overfitting, so the discretiser parameters need to be tuned
+under cv.
+
+After discretization, linear regression and decision tree make exactly the
+same prediction. As features are constant within each bin, any model must
+predict the same value for all points within a bin. Compared with the result
+before discretization, linear model become much more flexible while decision
+tree gets much less flexible. Note that binning features generally has no
+beneficial effect for tree-based models, as these models can learn to split
+up the data anywhere.
+
+"""
+
+# Author: Andreas Müller
+#         Hanmin Qin <qinhanmin2005@sina.com>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import KBinsDiscretizer
+from sklearn.tree import DecisionTreeRegressor
+
+print(__doc__)
+
+# construct the dataset
+rnd = np.random.RandomState(42)
+X = rnd.uniform(-3, 3, size=100)
+y = np.sin(X) + rnd.normal(size=len(X)) / 3
+X = X.reshape(-1, 1)
+
+# transform the dataset with KBinsDiscretizer
+enc = KBinsDiscretizer(n_bins=10, encode='onehot')
+X_binned = enc.fit_transform(X)
+
+# predict with original dataset
+fig, axes = plt.subplots(ncols=2, sharey=True, figsize=(10, 4))
+plt.sca(axes[0])
```

Comment:
Do we use sca much in other examples? It seems a bit unconventional. Then again, I can see how using axes methods together with pyplot.subplots might be seen as inconsistent